src/os/bluestore/NVMEDevice.cc

```diff
@@ -470,6 +478,11 @@ void SharedDriverQueueData::_aio_thread()
     }
 
     for (; t; t = t->next) {
+      if (current_queue_depth == max_queue_depth) {
+        // no slots
+        goto again;
```

Comment:
I think this is a little rough. do we need to introduce backoff method to avoid cpu spining?

Reply:
yes, I think so ... similar as what aio.cc does... @optimistyzy could you modify it?